lib/index.js

```diff
@@ -427,6 +487,37 @@ api.checkPermission = function(actor, permission, options, callback) {
     });
 };
 
+function _checkMembershipUpdate(actor, identity, groups, callback) {
+  // EXTERNAL: check to make sure IDENTITY_UPDATE_MEMBERSHIP permissions
+  //           exists for actor
+  // for each group in groups
+  //  check to make sure group exists
+  //  check to make sure group is a type: Group
+  //  x check to make sure owner is actor OR
+  //      actor has IDENTITY_UPDATE_MEMBERSHIP permission on resource??
+  async.each(groups, (group, callback) => {
+    async.auto({
+      get: callback => {
+        const query = {
+          id: database.hash(group),
+          'identity.type': 'Group',
+          'identity.sysStatus': 'active'
+        };
+        database.collections.identity.findOne(query, {
+          'identity.id': 1,
+          'identity.owner': 1
+        }, (err, record) => callback(err, record ? record.identity : null));
+      },
+      checkPermission: ['get', (callback, results) => {
+        // FIXME: is this permission check correct?
+        api.checkPermission(
+          actor, PERMISSIONS.IDENTITY_UPDATE_MEMBERSHIP,
+          {resource: results.get, translate: 'owner'}, callback);
```

Comment:
I'm going to need to chat with you about how we're going to implement "Add member to group", and how it may affect this code. Effectively, how do we do the multi-step "add person to group" feature? It's a larger conversation that may affect this code.